From 2abf514e93ee6f6b326ea7a23b7e51a140b797c5 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Thu, 18 Jan 2024 16:27:55 +0100
Subject: [PATCH 0569/1695] plat-stm32mp2: add A35 standardized status and
 control registers

Add A35 register definition to configure and control the PLL1.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e8dbfd1e7ccc62c0fcb41d90f4596c55dfe6fca5 plat-stm32mp2: add A35 standardized status and control registers]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ib7e5e4c125469e7fd3775a1ecbda20f315a3654a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/353997
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/stm32_sysconf.h | 48 +++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/stm32_sysconf.h b/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
index 954c02f4d..3e86cd779 100644
--- a/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
+++ b/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
@@ -37,6 +37,11 @@ enum syscon_banks {
 /*
  * A35SSC register offsets (base relative)
  */
+#define A35SS_SSC_CHGCLKREQ	SYSCON_ID(SYSCON_A35SSC, 0x0U)
+#define A35SS_SSC_PLL_FREQ1	SYSCON_ID(SYSCON_A35SSC, 0x80U)
+#define A35SS_SSC_PLL_FREQ2	SYSCON_ID(SYSCON_A35SSC, 0x90U)
+#define A35SS_SSC_PLL_EN	SYSCON_ID(SYSCON_A35SSC, 0xA0U)
+
 #define A35SSC_M33CFG_ACCESS_CR  SYSCON_ID(SYSCON_A35SSC, 0x2088U)
 #define A35SSC_M33_TZEN_CR       SYSCON_ID(SYSCON_A35SSC, 0x20A0U)
 #define A35SSC_M33_INITSVTOR_CR  SYSCON_ID(SYSCON_A35SSC, 0x20A4U)
@@ -53,6 +58,49 @@ enum syscon_banks {
  */
 #define A35SSC_M33_TZEN_CR_CFG_SECEXT          BIT(0)
 
+/*
+ * A35SSC A35SS_SSC_CHGCLKREQ register
+ */
+#define A35SS_SSC_CHGCLKREQ_ARM_CHGCLKREQ_EN	BIT(0)
+#define A35SS_SSC_CHGCLKREQ_ARM_CHGCLKREQ_MASK	BIT(0)
+
+#define A35SS_SSC_CHGCLKREQ_ARM_CHGCLKACK_MASK	BIT(1)
+#define A35SS_SSC_CHGCLKREQ_ARM_CHGCLKACK_SHIFT	U(1)
+
+/*
+ * A35SSC A35SS_SSC_PLL_FREQ1 register
+ */
+#define A35SS_SSC_PLL_FREQ1_FBDIV_MASK		GENMASK_32(11, 0)
+#define A35SS_SSC_PLL_FREQ1_FBDIV_SHIFT		U(0)
+
+#define A35SS_SSC_PLL_FREQ1_REFDIV_MASK		GENMASK_32(21, 16)
+#define A35SS_SSC_PLL_FREQ1_REFDIV_SHIFT	U(16)
+
+#define A35SS_SSC_PLL_FREQ1_MASK	(A35SS_SSC_PLL_FREQ1_REFDIV_MASK | \
+					 A35SS_SSC_PLL_FREQ1_FBDIV_MASK)
+
+/*
+ * A35SSC A35SS_SSC_PLL_FREQ2 register
+ */
+#define A35SS_SSC_PLL_FREQ2_POSTDIV1_MASK	GENMASK_32(2, 0)
+#define A35SS_SSC_PLL_FREQ2_POSTDIV1_SHIFT	U(0)
+
+#define A35SS_SSC_PLL_FREQ2_POSTDIV2_MASK	GENMASK_32(5, 3)
+#define A35SS_SSC_PLL_FREQ2_POSTDIV2_SHIFT	U(3)
+
+#define A35SS_SSC_PLL_FREQ2_MASK		GENMASK_32(5, 0)
+
+/*
+ * A35SSC A35SS_SSC_PLL_EN register
+ */
+#define A35SS_SSC_PLL_ENABLE_PD_EN			BIT(0)
+#define A35SS_SSC_PLL_ENABLE_PD_MASK			BIT(0)
+
+#define A35SS_SSC_PLL_ENABLE_LOCKP_MASK			BIT(1)
+
+#define A35SS_SSC_PLL_ENABLE_NRESET_SWPLL_FF_EN		BIT(2)
+#define A35SS_SSC_PLL_ENABLE_NRESET_SWPLL_FF_MASK	BIT(2)
+
 void stm32mp_syscfg_write(uint32_t id, uint32_t value, uint32_t bitmsk);
 uint32_t stm32mp_syscfg_read(uint32_t id);
 
-- 
2.47.2

